src/storage/timeval.c

```diff
@@ -33,78 +33,59 @@
 #include "timeval.h"
 
 gint
-timeval_is_older(GTimeVal t1, GTimeVal t2)
+timeval_is_older(GDateTime *t1, GDateTime *t2)
 {
-    if (t1.tv_sec > t2.tv_sec) {
-        return 1;
-    }
-
-    if (t1.tv_sec < t2.tv_sec) {
-        return -1;
-    }
-
-    if (t1.tv_usec == t2.tv_usec) {
-        return 0;
-    }
-
-    if (t1.tv_usec < t2.tv_usec) {
-        return -1;
-    }
-
-    return 1;
+    return g_date_time_compare(t1, t2);
 }
 
 const gchar *
-timeval_to_date(GTimeVal time, gchar *out)
+timeval_to_date(GDateTime *time, gchar *out)
 {
     GDate date;
-    g_date_set_time_val(&date, &time);
+    g_date_set_time_t(&date, g_date_time_to_unix(time));
     g_date_strftime(out, 11, "%Y/%m/%d", &date);
     return out;
 }
 
 
 const gchar *
-timeval_to_time(GTimeVal time, gchar *out)
+timeval_to_time(GDateTime *time, gchar *out)
 {
-    GDateTime *datetime = g_date_time_new_from_timeval_local(&time);
     g_sprintf(out, "%s.%06d",
-              g_date_time_format(datetime, "%H:%M:%S"),
-              (guint) time.tv_usec
+              g_date_time_format(time, "%H:%M:%S"),
+              g_date_time_get_microsecond(time)
     );
-    g_date_time_unref(datetime);
     return out;
 }
 
 const gchar *
-timeval_to_duration(GTimeVal start, GTimeVal end, gchar *out)
+timeval_to_duration(GDateTime *start, GDateTime *end, gchar *out)
 {
     g_return_val_if_fail(out != NULL, NULL);
 
-    if (start.tv_sec == 0 || end.tv_sec == 0)
+    if (NULL == start || NULL == end)
```

Comment:
I'll try to avoid mixing traditional conditions with [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) for now